public/app/features/explore/utils/links.ts

```diff
@@ -72,22 +72,37 @@ export const getFieldLinksForExplore = (options: {
     text: 'Raw value',
   };
 
+  let fieldDisplayValuesProxy;
+
   // If we have a dataFrame we can allow referencing other columns and their values in the interpolation.
   if (dataFrame) {
+    fieldDisplayValuesProxy = getFieldDisplayValuesProxy({
+      frame: dataFrame,
+      rowIndex,
+    });
+
     scopedVars['__data'] = {
       value: {
         name: dataFrame.name,
         refId: dataFrame.refId,
-        fields: getFieldDisplayValuesProxy({
-          frame: dataFrame,
-          rowIndex,
-        }),
+        fields: fieldDisplayValuesProxy,
       },
       text: 'Data',
     };
   }
 
   if (field.config.links) {
+    if (fieldDisplayValuesProxy && fieldDisplayValuesProxy[field.name] !== undefined) {
+      scopedVars['__targetField'] = {
+        value: fieldDisplayValuesProxy[field.name],
+        text: fieldDisplayValuesProxy[field.name],
+      };
+      scopedVars[field.name] = {
```

Comment:
This will allow to access only the target field. I think we need to loop over all fields to allow accessing any field value:

```typescript
dataFrame.fields.forEach((f) => {
  scopedVars[f.name] = fieldDisplayValuesProxy[f.name];
});
```

`links.test.ts` needs to be updatef because at the moment it uses only one field in the setup.

Reply:
@ifrost I think this function is called once per field, so we only need to add the var for the field that calls it.

Reply:
Okay, I changed it so each field has a variable for every other field in the dataframe. I added a test for this as well.